yt/yt/server/queue_agent/queue_static_table_exporter.cpp

```diff
@@ -770,63 +1039,76 @@ TQueueExporter::TQueueExporter(
-    , Executor_(New<TScheduledExecutor>(
-            Invoker_,
-            BIND_NO_PROPAGATE(&TQueueExporter::Export, MakeWeak(this)),
+    , Executor_(New<TPeriodicExecutor>(
+            invoker,
+            BIND_NO_PROPAGATE(&TQueueStaticTableExporter::Pass, MakeWeak(this)),
             /*interval*/ std::nullopt
         ))
     , Logger(QueueStaticTableExporterLogger().WithTag("%v, ExportName: %v",
         logger.GetTag(),
         ExportName_))
 {
     Executor_->Start();
-    Executor_->SetInterval(DynamicConfig_.Enable
-        ? std::optional(ExportConfig_.ExportPeriod)
+    Executor_->SetPeriod(DynamicConfig_.Enable
+        ? std::optional(DynamicConfig_.PassPeriod)
         : std::nullopt);
 }
 
-void TQueueExporter::Reconfigure(const TQueueStaticExportConfig& newExportConfig, const TQueueExporterDynamicConfig& newDynamicConfig)
+void TQueueStaticTableExporter::OnExportConfigChanged(const TQueueStaticExportConfigPtr& newExportConfig)
 {
     auto guard = Guard(Lock_);
 
-    if (ExportConfig_.ExportDirectory != newExportConfig.ExportDirectory) {
+    if (ExportConfig_->ExportDirectory != newExportConfig->ExportDirectory) {
         ExportProgress_ = New<TQueueExportProgress>();
     }
 
-    if (DynamicConfig_.Enable != newDynamicConfig.Enable || ExportConfig_.ExportPeriod != newExportConfig.ExportPeriod) {
-        Executor_->SetInterval(newDynamicConfig.Enable
-            ? std::optional(newExportConfig.ExportPeriod)
+    ExportConfig_ = newExportConfig;
+}
+
+void TQueueStaticTableExporter::OnDynamicConfigChanged(const TQueueExporterDynamicConfig& newDynamicConfig)
+{
+    auto guard = Guard(Lock_);
+
+    if (DynamicConfig_ != newDynamicConfig) {
+        Executor_->SetPeriod(newDynamicConfig.Enable
+            ? std::optional(newDynamicConfig.PassPeriod)
             : std::nullopt);
     }
 
-    ExportConfig_ = newExportConfig;
     DynamicConfig_ = newDynamicConfig;
 }
 
-TQueueExportProgressPtr TQueueExporter::GetExportProgress() const
+TQueueExportProgressPtr TQueueStaticTableExporter::GetExportProgress() const
 {
     auto guard = Guard(Lock_);
     return ExportProgress_;
 }
 
-NQueueClient::TQueueStaticExportConfig TQueueExporter::GetConfig()
+//! Pass that is executed regularly to check whether an export can be started.
+//!
+//! Default period is a minute, so by default between consecutive export iterations there is at least 60 seconds delay.
+//! Furthermore, all exports within one instance share the same throttler to limit number of exports started per second.
+void TQueueStaticTableExporter::Pass()
 {
-    auto guard = Guard(Lock_);
-    return ExportConfig_;
-}
+    auto traceContextGuard = TTraceContextGuard(TTraceContext::NewRoot("QueueExporterPass"));
 
-void TQueueExporter::Export()
-{
-    VERIFY_INVOKER_AFFINITY(Invoker_);
+    auto now = TInstant::Now();
+
+    auto guard = Guard(Lock_);
+    auto exportUnixTs = GetExportFragmentUnixTs(now.Seconds(), ExportConfig_->ExportPeriod);
+    guard.Release();
```

Comment:
Let's introduce some method for getting the whole config under lock, and use it here and in other places to avoid manually releasing guards. But you will need to deep-copy, since dynamic config changes can happen.